### PR TITLE
Add para assignment dialog

### PR DESCRIPTION
**Pull request by codetheweb**
Closes https://github.com/sfbrigade/compass/issues/236.

Demo:

https://github.com/sfbrigade/compass/assets/7410405/0b374b13-3899-45de-a05b-7e439b6596d5

### CODE REVIEW

File: src/components/subgoal/Subgoal-Assignment-Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  };
+
+  return (
+    <Dialog open={props.isOpen} onClose={handleClose}>
+      <DialogTitle>Assign to benchmark</DialogTitle>
+
+      <Box sx={{ px: 2, pb: 2 }}>
+        <Alert severity="info" sx={{ mb: 2 }}>
+          <AlertTitle>Benchmark</AlertTitle>
+          By October, when given a list of 20 unfamiliar words that contain
```
Comment (codetheweb) [inline src/components/subgoal/Subgoal-Assignment-Modal.tsx:49]: where can we pull this from?

Comment (tessathornberry) [reply src/components/subgoal/Subgoal-Assignment-Modal.tsx:49]: Looks like the closest API we've made is iep.getSubgoalAndTrialData, for data collection, which we are using in /src/pages/benchmarks/[benchmark_id]/index.tsx, but we do not yet have an API for an individual Subgoal without the Trial data.

File: src/components/subgoal/Subgoal-Assignment-Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                  <ListItemText>
+                    {para.first_name} {para.last_name}
+                  </ListItemText>
+                </ListItemButton>
+              </ListItem>
+            ))}
+          </List>
+        </Box>
+        <Box sx={{ display: "flex", justifyContent: "end" }}>
+          <Button variant="contained" onClick={handleClose}>
```
Comment (tessathornberry) [inline src/components/subgoal/Subgoal-Assignment-Modal.tsx:84]: Great start!

File: src/components/subgoal/Subgoal.tsx [hunk 3]
```diff
@@ -2,25 +2,17 @@ import { trpc } from "@/client/lib/trpc";
...
 
-  // TODO: add form to assign to my paras
-  // const assignToPara = (event: React.FormEvent<HTMLFormElement>) => {
-  //   event.preventDefault();
-  //   const data = new FormData(event.currentTarget);
-
-  //   task.mutate({
-  //     subgoal_id: subgoal.subgoal_id,
-  //     assignee_id: data.get("assignee_id") as string,
-  //     due_date: new Date(data.get("due_date")) as Date
```
Comment (tessathornberry) [inline src/components/subgoal/Subgoal.tsx:21]: Let's be sure to test this when we implement it, to ensure it comes through as the correct date.

**Review by tessathornberry — APPROVED** [review]
This is a great starting point. Thank you, Max!

File: src/components/subgoal/Subgoal-Assignment-Modal.tsx (hunk 1 above)
Comment (tessathornberry) [reply src/components/subgoal/Subgoal-Assignment-Modal.tsx:49]: We also have iep.tempAddTaskToSelf for assignment,being used in the Subgoal.tsx

**Pull request merged**